src/GovUk.Education.ExploreEducationStatistics.Common/Services/Interfaces/IInMemoryCacheService.cs

```diff
@@ -0,0 +1,69 @@
+﻿#nullable enable
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using GovUk.Education.ExploreEducationStatistics.Common.Cache.Interfaces;
+using static System.Linq.Enumerable;
+
+namespace GovUk.Education.ExploreEducationStatistics.Common.Services.Interfaces;
+
+/**
+ * Interface for services that handle in-memory caches.
+ *
+ * There is deliberately no `Delete` method defined on this interface, as the client of this service should instead be
+ * setting rules around cache expiry of the items being cached.
+ *
+ * The most important reason that there is no `Delete` method for use is that this in-memory caching strategy is
+ * not distributed in any way and so a Delete operation invoked on a cache on a particular server instance would have no
+ * effect on any additional servers running the same service elsewhere e.g. in a horizontally scaled service.
+ *
+ * Items that are subject to regular updates are therefore advised to be cached for a very short-lived timeframe so as
+ * to prevent consumers from having to wait too long before the updated content is available.  
+ */
+public interface IInMemoryCacheService
+{
+    Task<object?> GetItem(IInMemoryCacheKey cacheKey, Type targetType);
+
+    Task SetItem<TItem>(
+        IInMemoryCacheKey cacheKey, 
+        TItem item, 
+        InMemoryCacheConfiguration configuration,
+        DateTime? nowUtc = null);
+}
+
+/**
+ * This enum allows us to specify that, in addition to the cache duration requested when setting an item in the cache,
+ * the specified cache duration will be truncated so as to not exceed certain regular points of time during the day.
+ *
+ * As an example, if an item is being set in the cache at 2:27pm with a requested cache duration of 5 minutes, it would
+ * expire at 2:32pm assuming that no ExpirySchedule was also requested when setting that item.
+ *
+ * However, if an item is being set in the cache at 2:27pm with a requested cache duration of 5 minutes AND an
+ * ExpirySchedule of HalfHourly, then the cache duration will be truncated to expire the item at 2:30pm (i.e. on the
+ * half hour).
+ *
+ * This allows us to request reasonably-sized cache durations for our items, but allowing us to expire them at certain
+ * critical points during the day where we expect these items to potentially change.
+ */
+public enum ExpirySchedule
```

Comment:
Yes!  The Cron library made a very neat solution rather than hand-rolling, thanks!

In addition to swapping out the hand-cranked time periods code, I've also made the timeouts configurable via appsettings, to reduce the cached times in environments where we run UI tests - this is so that we do not need to wait longer than we realistically need to in order to see changes reflected in the public site after updating in admin.

Note that the GetLatestRelease cache duration is smaller than the GetRelease timeout.  This is in order to get changes in a newly published Release for a Publication out as quickly as possible for people hitting the Publication Slug URL directly, as they might be at 930am on release day (although prior investigations suggested that most viewers actually came in via the publication-slug/release-slug URL which, for a brand new Release, would have no prior cached version anyway, and so can tolerate a higher cache timeout).

Reply:
I see @ntsim suggested using Cronos or NCrontab and we've chosen to go with Cronos. We already use NCrontab elsewhere in the project, so just wondering if you were aware of that and if there are any benefits of one over the other? It would be nice to use the same library if possible.

Reply:
Good point!  For now I have replaced Cronos with Ncrontab for consistency with the Publisher code.